Sources/ProjectDescription/ProjectOptions.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+
+extension Project {
+    /// Additional options related to the `Project`
+    public struct Options: Codable, Equatable {
+        /// Defines how to generate automatic schemes
+        public let automaticSchemesOptions: AutomaticSchemesOptions
+
+        /// Disables generating Bundle accessors.
+        public let disableBundleAccessors: Bool
+
+        /// Disable the synthesized resource accessors generation
+        public let disableSynthesizedResourceAccessors: Bool
+
+        /// Text settings to override user ones for current project
+        public let textSettings: TextSettings
+
+        public static func options(
+            automaticSchemesOptions: AutomaticSchemesOptions = .enabled(),
+            disableBundleAccessors: Bool = false,
+            disableSynthesizedResourceAccessors: Bool = false,
+            textSettings: TextSettings = .textSettings()
+        ) -> Self {
+            self.init(
+                automaticSchemesOptions: automaticSchemesOptions,
+                disableBundleAccessors: disableBundleAccessors,
+                disableSynthesizedResourceAccessors: disableSynthesizedResourceAccessors,
+                textSettings: textSettings
+            )
+        }
+    }
+}
+
+// MARK: - AutomaticSchemesOptions
+
+extension Project.Options {
+    /// The automatic schemes options
+    public enum AutomaticSchemesOptions: Codable, Equatable {
+        /// Defines how to group targets into scheme for autogenerated schemes
+        public enum TargetSchemesGrouping: Codable, Equatable {
```

Comment:
Perhaps these should be a `struct` as well? As it is you can only choose a single scheme grouping option, but I can see value in allowing selection of multiple options.

Reply:
Not sure I got it. Do you mean a different strategy for different projects?

Reply:
No, for a single project. Suppose I want grouped schemes, but I also want a scheme which includes everything.

Reply:
I wouldn't add it because:
1. you still have the `-Workspace` scheme including the whole workspace
2. if you really need it you can still create it "manually" in Project.swift